Mathlib/Topology/Compactness/Lindelof.lean

```diff
@@ -177,6 +176,29 @@ theorem IsLindelof.elim_nhds_subcover (hs : IsLindelof s) (U : X → Set X)
   · have : ⋃ x ∈ t, U ↑x = ⋃ x ∈ Subtype.val '' t, U x := biUnion_image.symm
     rwa [← this]
 
+/-- For every nonempty open cover of a Lindelöf set, there exists a subcover indexed by ℕ. -/
+theorem IsLindelof.indexed_countable_subcover {ι : Type v} [Nonempty ι]
+    (hs : IsLindelof s) (U : ι → Set X) (hUo : ∀ i, IsOpen (U i)) (hsU : s ⊆ ⋃ i, U i) :
+    ∃ f : ℕ → ι, s ⊆ ⋃ n, U (f n) := by
+  rcases IsLindelof.elim_countable_subcover hs U hUo hsU with ⟨c, ⟨c_count, c_cov⟩⟩
+  wlog c_nonempty : c.Nonempty
+  · simp only [not_nonempty_iff_eq_empty.mp c_nonempty, mem_empty_iff_false, iUnion_of_empty,
+      iUnion_empty] at c_cov
+    rcases (exists_const ι).mpr trivial with ⟨i⟩
+    use fun _ ↦ i
+    rw [subset_eq_empty c_cov rfl]
+    exact empty_subset _
```

Comment:
```suggestion
    simp only [subset_eq_empty c_cov rfl, empty_subset, exists_const]
```

Reply:
Yep, but I want to try it out myself first.

Reply:
Sure, even better! I just wanted to make sure that you knew how to proceed. Apologies for the triviality ;-)